x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapper.java

```diff
@@ -368,6 +437,22 @@ protected void index(DocumentParserContext context, Geometry geometry) throws IO
         }
     }
 
+    @Override
+    protected void indexScriptValues(
+        SearchLookup searchLookup,
+        LeafReaderContext readerContext,
+        int doc,
+        DocumentParserContext documentParserContext
+    ) {
+        fieldType().scriptValues.valuesForDoc(searchLookup, readerContext, doc, geometry -> {
+            try {
+                index(documentParserContext, geometry);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);  // only thrown by MultiFields which is always null
```

Comment:
I think this is actually thrown by `WellKnownBinary.toWKB`?  But looking at that method it only gets thrown from `ByteArrayOutputStream.close()`, which is actually a no-op and can safely be ignored.  So I think I would move the exception handling there.

It's a bit annoying that `ByteArrayOutputStream.close()` declares that it throws an IOException really.  Tsk, a bug in the java API.

Reply:
I am opening a different PR for this suggestion as it will touche other unrelated parts of the code.

Reply:
https://github.com/elastic/elasticsearch/pull/100669

Reply:
Updated